pkg/dispatcher/operator/workers/controller.go

```diff
@@ -101,6 +101,26 @@ func NewController(ctx context.Context, kubeClient kubernetes.Interface, podInfo
 	return controller
 }
 
+// Utility function
+func (c *Controller) deleteCalculation(pod *corev1.Pod, name string, key string) error {
+	listOfCalculations, _ := c.calculationLister.List(labels.Set{"assign": name}.AsSelector())
+	calc, err := hasAssignedCalculation(listOfCalculations)
+
+	toUpdate := make(map[string]interface{})
+	toUpdate["status"] = ""
+
+	if !err {
+		return fmt.Errorf("error temporary solution")
+	}
+
+	if calc.Phase == calculationsv1.ProcessingPhase && pod.Name == name {
```

Comment:
`pod.Name` and `name` will always be the same here.

Reply:
if I had this code to iterate through the calculations: for _, c := range calculations {}
would pod.Name == c.Name help?

Reply:
`pod.Name` refers to the name of the pod and the `c.Name` refers to the calculation name. Those are two different things.